File: ejerc.diccionarios/Ejercicio 3/Ejercicio3.java
// Importa las clases necesarias para trabajar con mapas y entrada de datos
import java.util.HashMap;
import java.util.Map;
import java.util.Scanner;

public class RepuestosCarro {
    public static void main(String[] args) {
        // Crea un mapa (diccionario) que asocia el nombre del repuesto con su precio
        Map<String, Double> repuestos = new HashMap<>();

        // Agrega algunos repuestos con su respectivo precio
        repuestos.put("Bujia", 1.40);
        repuestos.put("Pastillas de freno", 5.80);
        repuestos.put("Radiador", 300.50);
        repuestos.put("Rodamiento", 10.70);

        // Crea un objeto Scanner para leer datos ingresados por el usuario
        Scanner sc = new Scanner(System.in);

        // Solicita al usuario el nombre del repuesto que desea consultar
        System.out.print("Ingrese el nombre del repuesto: ");
        String repuesto = sc.nextLine().trim(); // Lee el texto y elimina espacios innecesarios

        // Solicita la cantidad del repuesto que el usuario desea comprar
        System.out.print("Ingrese la cantidad de " + repuesto + ": ");
        int cantidad = sc.nextInt(); // Lee el número entero ingresado

        // Verifica si el repuesto ingresado existe en el mapa
        if (repuestos.containsKey(repuesto)) {
            // Calcula el costo total multiplicando el precio unitario por la cantidad
            double total = repuestos.get(repuesto) * cantidad;

            // Muestra el costo total con formato de dos decimales
            System.out.printf("El costo total de %d %s(s) es: %.2f dólares.%n", cantidad, repuesto, total);
        } else {
            // Si el repuesto no está en el mapa, muestra un mensaje de error
            System.out.println("El repuesto '" + repuesto + "' no está en el diccionario.");
        }

        // Cierra el objeto Scanner para liberar recursos
        sc.close();
    }
}

File: ejerc.diccionarios/Ejercicio 3/Ejercicio3.py
# Diccionario con los precios de los repuestos
repuestos = {
    'Bujia': 1.40,               # Precio de una bujía en dólares
    'Pastillas De Freno': 5.80,  # Precio de un juego de pastillas de freno en dólares
    'Radiador': 300.50,          # Precio de un radiador en dólares
    'Rodamiento': 10.70          # Precio de un rodamiento en dólares
}

# Solicitar al usuario el nombre del repuesto
# .strip() elimina espacios extra al inicio y al final
# .title() convierte el texto a formato capitalizado, para que coincida con las claves del diccionario
repuesto = input("Ingrese el nombre del repuesto: ").strip().title()

# Solicitar al usuario la cantidad de repuestos que quiere comprar, y convertirla a entero
cantidad = int(input(f"Ingrese la cantidad de {repuesto}: "))

# Verificar si el repuesto existe en el diccionario
if repuesto in repuestos:
    # Calcular el costo total multiplicando el precio unitario por la cantidad solicitada
    total = repuestos[repuesto] * cantidad
    # Mostrar el resultado con dos decimales
    print(f"El costo total de {cantidad} {repuesto}(s) es: {total:.2f} dólares.")
else:
    # Mensaje si el repuesto no está disponible en el diccionario
    print(f"El repuesto '{repuesto}' no está en el diccionario.")
